src/common/PreviewUtils.ts

```diff
@@ -128,4 +129,44 @@ export class PreviewUtils {
             return undefined;
         }
     }
+
+    /**
+     * For the new Local Development Preview approach where a new local dev server is being used,
+     * the local dev server needs to be configured to use a web socket url for local previewing.
+     * This method generates an appropriate web socket url to be used by the local dev server.
+     *
+     * @param platform a platform flag (desktop , ios, android).
+     * @param port the port number that the local dev server is configured to use.
+     * @returns A string representing a web socket url to be used by the local dev server.
+     */
+    public static generateWebSocketUrlForLocalDevServer(platform: string, port: string): string {
+        /*
+          - For desktop browsers other than Safari, local development use cases will target ws://localhost:<port> connections to the local dev server
+          - For the Safari desktop browser, target wss://localhost:<port>
+            
+          - For mobile (webview in native apps), local development use cases will target:
+            - iOS: wss://localhost:<port>
+            - Android: wss://10.0.2.2:<port>
+        */
+
+        if (CommandLineUtils.platformFlagIsIOS(platform)) {
+            return `wss://localhost:${port}`;
+        }
+
+        if (CommandLineUtils.platformFlagIsAndroid(platform)) {
+            return `wss://10.0.2.2:${port}`;
+        }
+
+        if (process.platform !== 'darwin') {
+            return `ws://localhost:${port}`; // cannot be Safari since it is only available on Mac
+        }
+
+        // Check to see if the default browser is Safari
+        // From https://apple.stackexchange.com/questions/313454/applescript-find-the-users-set-default-browser
+        const cmd =
+            "defaults read ~/Library/Preferences/com.apple.LaunchServices/com.apple.launchservices.secure | awk -F'\"' '/http;/{print window[(NR)-1]}{window[NR]=$2}'";
```

Comment:
The thing I'm not wild about with this approach is making `awk` do the post-processing—mostly because of how it's not easy to read, maintain, or test (as `awk` tends to be).

I wonder if we shouldn't use something like [`plutil`](https://stackoverflow.com/a/63790334) to put what otherwise comes out of the `defaults read` output into a parseable format, and parse it ourselves? May be overkill though.

Reply:
As you said, it might be overkill. I thought about using packages like [default-browser-id](https://www.npmjs.com/package/default-browser-id) or [default-browser](https://www.npmjs.com/package/default-browser) but thought that those too might be over kill?

We only really care about whether on macOS the default browser is Safari or not. Looking at [its code](https://www.npmjs.com/package/default-browser-id?activeTab=code), `default-browser-id` is the most light-weight one but still, is it worth pulling this one in vs just use `awk` ?

We could always update the logic here to use something else in the future if need be?

Reply:
> We could always update the logic here to use something else in the future if need be?

Let's go with that. It's a minor concern, and mostly due to the fact that it's not the greatest insight into the plist data that's returned. But there's not much else to be gained from that plist data for our use case, anyway, so... ¯\\_(ツ)_/¯ yeah let's punt.